pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -93,7 +95,10 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 	}
 	controllerutil.RemoveFinalizer(nodeClaim, v1beta1.TerminationFinalizer)
 	if !equality.Semantic.DeepEqual(stored, nodeClaim) {
-		if err = c.kubeClient.Patch(ctx, nodeClaim, client.MergeFrom(stored)); err != nil {
+		if err = c.kubeClient.Update(ctx, nodeClaim); err != nil {
+			if errors.IsConflict(err) {
```

Comment:
Why are we requiring only on conflict errors?

Reply:
We're requeueing on other errors too, see L102. This logic adds handling so a conflict error on an Update isn't treated as a true controller error